LinqAssignment/Services/LinqQueiries.cs

```diff
@@ -0,0 +1,192 @@
+﻿using Linq.Data.Models;
+
+namespace LinqAssignment.Services
+{
+    public class LinqQueiries
+    {
+        List<Country> countries;
+        public LinqQueiries()
+        {
+            DataGenerator dataGenerator = new DataGenerator();
+
+            countries = dataGenerator.GetCountries();
+        }
+
+        public Country AddCountry()
+        {
+            // Add a new Country called "MyOwn"
+            var myOwnCountry = new Country { CountryId = 4, CountryName = "MyOwn" };
+            countries.Add(myOwnCountry);
+
+            // Add a range of 10 states to "MyOwn" country with each state having 10 districts
+            for (int i = 1; i <= 10; i++)
+            {
+                var state = new State { StateId = 100 + i, StateName = $"State{i}", CountryId = 4 };
+                myOwnCountry.States.Add(state);
+
+                for (int j = 1; j <= 10; j++)
+                {
+                    var district = new District
+                    {
+                        DistrictId = (i - 1) * 10 + j + 1000,
+                        DistrictName = $"State{i}District{j}",
+                        StateId = 100 + i,
+                        DistrictPopulation = new Random().Next(1000, 10000)
+                    };
+                    state.Districts.Add(district);
+                }
+            }
+            return myOwnCountry;
+        }
+        public List<Country> GetAllCountries()
+        {
+            return countries;
+        }
+        public List<State> GetAllStatesofCountry()
+        {
+            return countries.SelectMany(c => c.States).ToList();
```

Comment:
instead of **"countries.SelectMany(c => c.States).ToList()"** can we use  **"countries.States.ToList()"**  directly.